src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,101 @@
+package duke;
+
+/**
+ * Parser class is an abstraction of an object that parses and interprets text messages into commands.
+ */
+
+public class Parser {
+    public TaskList taskList;
+    public boolean isAlive;
+
+    /**
+     * Instantiates the Parser with attributes.
+     * @param taskList the task list with a list of commands
+     */
+    public Parser (TaskList taskList) {
+        this.taskList = taskList;
+        this.isAlive = true;
+    }
+
+    /**
+     * This parses and executes relevant comments from the tasklist
+     * @param command the command given by the user
+     * @throws DukeException an exception due to errors in parsing text
+     */
+    public void executeCommand(String command) throws DukeException {
```

Comment:
In the Parser.java, the executeCommand method consists of more than 50 lines of code. One piece of advice I will like to share with you is to try to avoid long methods and abide closely by the 30 lines of code rule for a method.